.github/workflows/main.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 13.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: '13.x'
+      - name: Cache Node.js modules
+        uses: actions/cache@v1
+        with:
+          path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - run: npm ci
+      - run: npm test
+        env:
+          CI: 'true'
+  # build image for docker hub (kwizapp/)
+  build_docker_hub:
+    runs-on: ubuntu-latest
+    needs: test # required that this step has completed
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: azure/docker-login@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: Build the Docker image
+        run: docker build . --tag kwizapp/nest-api:latest
+      - name: Push the Docker image to Docker Hub
```

Comment:
```suggestion
      - name: Push the Docker image to Docker Hub
        if: github.branch == 'dev' || github.branch == 'master'
```

Reply:
the latest image on docker hub should always be the latest state from our dev branch. we want to run a docker build for every commit to ensure that images can be built. however, we only want to override the docker hub image if we are on dev or master.

Reply:
good catch! thanks for that :)